camayoc/tests/remote/quipucords/test_scan.py

```diff
@@ -0,0 +1,341 @@
+        return cfg.get('qcs', {}).get('scans', [])
+    except (ConfigFileNotFoundError, KeyError):
+        return []
+
+
+def get_scan_result(scan_name):
+    """Collect scan results from global cache.
+
+    Raise an error if no results are available, because this means the scan
+    was not even attempted.
+    """
+    result = SCAN_DATA.get(scan_name)
+    if result is None:
+        raise RuntimeError(
+            'Absolutely no results available for scan named {scan_name},'
+            'because no scan was even attempted.'.format(
+                scan_name=scan_info['name'])
+        )
+    return result
+
+
+@pytest.mark.parametrize(
+    'scan_info', scan_info(), ids=utils.name_getter)
+def test_scan_complete(scan_info):
+    """Test each scan report.
+
+    :id: 4e7a608e-a18d-48db-b263-a0d5474385dd
+    :description: Test if the scan completed.
+    :steps: Check the final status of the scan.
+    :expectedresults: Scans should complete and report their finished status.
+    """
+    result = get_scan_result(scan_info['name'])
+    if result['final_status'] != 'completed':
+        raise AssertionError(
+            'Scan did not complete. Its final status was {status}. It'
+            ' encountered the following errors: {errors}\n'.format(
+                status=result['final_status'],
+                errors=' '.join(result['errors']),
+            )
+
+        )
+
+
+@pytest.mark.parametrize(
+    'scan_info', scan_info(), ids=utils.name_getter)
+def test_scan_connection_results(scan_info):
+    """Test each scan report.
+
+    :id: 8087fdc9-6626-476a-a11a-4783cbf501a0
+    :description: Test the connection results of the scan
+    :steps: Inspect the connection results for any failures.
+    :expectedresults: The scan connected to each host.
+    """
+    result = get_scan_result(scan_info['name'])
+    assert result['connection_results'] is not None
+
+
+@pytest.mark.parametrize(
+    'scan_info', scan_info(), ids=utils.name_getter)
+def test_scan_inspection_results(scan_info):
+    """Test each scan report.
+
+    :id: d29d9fd6-ac85-4110-9b1c-ac7c26205d16
+    :description: Test the inspection results of the scan
+    :steps: Inspect the inspection results for any failures.
+    :expectedresults: The scan connected to each host.
+    """
+    result = get_scan_result(scan_info['name'])
+    assert result['inspection_results'] is not None
+
+
+@pytest.mark.skip
+@pytest.mark.parametrize(
+    'scan_info', scan_info(), ids=utils.name_getter)
+def test_scan_os_report(scan_info):
+    """Test each scan report.
+
+    :id: 326ceaef-5166-494e-a12c-fc3959fa1f9f
+    :description: Test the inspection results of the scan
+    :steps: Inspect the inspection results for any failures.
+    :expectedresults: The scan connected to each host.
+    """
+    pass
+
+
+@pytest.mark.skip
+@pytest.mark.parametrize(
+    'scan_info', scan_info(), ids=utils.name_getter)
+def test_scan_product_report(scan_info):
+    """Test each scan report.
```

Comment:
Same here, please make more clear what type of information you are looking for on the inspection_results.